ui/DesignSystem/Component/List.svelte

```diff
@@ -1,11 +1,12 @@
-<script>
+<script lang="ts">
   import { createEventDispatcher } from "svelte";
 
   const dispatch = createEventDispatcher();
 
-  export let dataCy = null;
-  export let items = null;
-  export let style = null;
+  export let items: any[];
+
+  export let dataCy: string | undefined = undefined;
```

Comment:
Not the prettiest syntax; an alternative could be 

```typescript
export let dataCy = "";
```

wherein the `string` type would be inferred, and we would check truthiness like `dataCy.length > 0`.

Reply:
_Maybe_, we could define a type for `dataCy` in general:
```ts
interface NoData {
  kind: "no-data";
}

interface Data {
  kind: "data";
  value: string;
}

type DataCy = Data | NoData
```

In general, does anybody know of any `Maybe`/`Optional` type in typescript? Or even one step further, is there any good collection of functional-esque resources as ts library?

Having a `Maybe` type would help with all the other `undefined` cases as well. For example, we would have:
```ts
let badge: Maybe<BadgeType> = Nothing;
```

Reply:
```suggestion
  export let dataCy = "";
```

There is [prelude-ts](https://github.com/emmanueltouzery/prelude-ts) which offers FP shenanigans and among them an `Option`. Maybe - no pun intended - we can start putting together a little prelude with some FP sprinkles.

It seems if a prop is optional it needs a default value, which would avoid presence tests alltogether.

Reply:
These also seem like good options:
* https://gigobyte.github.io/purify/
* https://gcanti.github.io/fp-ts/

`fp-ts` looks pretty complete :ok_hand: Might become a front end engineer if we use it because it has higher kinded types and the whole shebang of classes

Reply:
We could give it a spin if we have actual use-cases. In thise specific instance I believe a default solves it.

Reply:
Sure, but there were other cases where I saw `| undefined` was being used. If we get some breathing room I'd love to see someone spearhead a `radicle-decision` on a sanctified lib for it :)

But agreed, the default is fine here. `dataCy` is just magic to me anyway. Basically a macro in my eyes